analytics_data_api/v0/views/courses.py

```diff
@@ -627,36 +629,31 @@ class ProblemsListView(BaseCourseView):
             * correct_submissions: Total number of *correct* submissions.
             * part_ids: List of problem part IDs
     """
-    model = models.ProblemResponseAnswerDistribution
     serializer_class = serializers.ProblemSerializer
-
-    def apply_date_filtering(self, queryset):
-        # Date filtering is not possible for this data.
-        return queryset
+    allow_empty = False
 
     def get_queryset(self):
-        queryset = super(ProblemsListView, self).get_queryset()
-        queryset = queryset.order_by('module_id', 'part_id')
+        sql = """
+SELECT
+    module_id,
+    SUM(count) AS total_submissions,
+    SUM(CASE WHEN correct=1 THEN count ELSE 0 END) AS correct_submissions,
+    GROUP_CONCAT(DISTINCT part_id) AS part_ids,
+    MAX(created) AS created
+FROM answer_distribution
+WHERE course_id = %s
+GROUP BY module_id;
+        """
+        with connections[settings.ANALYTICS_DATABASE].cursor() as cursor:
+            cursor.execute(sql, [self.course_id])
+            rows = dictfetchall(cursor)
 
-        data = []
+        for row in rows:
+            # Convert the comma-separated list into an array of strings.
+            row['part_ids'] = row['part_ids'].split(',')
```

Comment:
The previous code had `part_ids = set()    # Use a set to remove duplicate values.`.  It seems this change may introduce duplicates.